nextcloudappstore/core/api/v1/views.py

```diff
@@ -167,3 +169,36 @@ def post(self, request, *args, **kwargs):
 
         token, created = Token.objects.get_or_create(user=user)
         return Response({'token': token.key})
+
+
+class RegenerateAuthToken(APIView):
+    """Generates a new API token for the authenticated user, regardless of
+    whether a token already exists.
+
+    Accepts TokenAuthentication and BasicAuthentication.
+    """
+
+    authentication_classes = (authentication.TokenAuthentication,
+                              authentication.BasicAuthentication,)
+    throttle_classes = (PostThrottle,)
+    permission_classes = (IsAuthenticated,)
+    parser_classes = (parsers.FormParser,
+                      parsers.MultiPartParser,
+                      parsers.JSONParser,)
+    renderer_classes = (renderers.JSONRenderer,)
+    serializer_class = AuthTokenSerializer
+
+    def post(self, request, *args, **kwargs):
+        if request.user.is_authenticated():
```

Comment:
No need to do these is_authenticated checks because permission_classes = (IsAuthenticated,) and if the user is not authenticated, it will be null and you cant update it